src/app/models/form_issue.server.model.ts

```diff
@@ -36,7 +40,22 @@ const FormIssueSchema = new Schema<IFormIssueSchema, IFormIssueModel>(
 FormIssueSchema.index({ formId: 1 })
 
 /**
- * Form Issue Schema
+ * Returns a cursor for all issues for the form with formId.
+ * @param formId the form id to return the cursor for
+ * @returns a cursor to the issue retrieved
+ */
+FormIssueSchema.statics.getIssueCursorByFormId = function (
+  formId: string,
+): QueryCursor<FormIssueStreamData> {
+  return this.find({ formId }, { issue: 1, email: 1, created: 1 })
+    .batchSize(2000)
+    .read('secondaryPreferred')
```

Comment:
Not exactly sure if there's any specific impact if we use `secondary`, but from other queries, it seemed like non-critical queries are using `secondary` instead of `sp`

Reply:
using secondary means if secondary not avail, we fail the request. i think we can use primary if secondary is not avail, since the query is indexed. Ref [here](https://www.mongodb.com/docs/manual/core/read-preference/#:~:text=Starting%20in%20version%204.4%2C%20secondary%20supports%20hedged%20reads%20on%20sharded%20clusters.&text=In%20most%20situations%2C%20operations%20read%20from%20secondary%20members%20but%20if,hedged%20reads%20on%20sharded%20clusters.)

Reply:
Yeah I know the difference, but any reason why we wanna deviate from using `secondary`? The query can be indexed, but the payload transfer will still impact it (although it is unlikely that it will be an issue here)

Reply:
Because I think it does not make sense to fail the query when primary is up. But if you think we should stick with `secondary` I am ok to change it

Reply:
I think that there's merit in such queries not affecting `primary`. But not a strong opinion! I'll leave the final say to you

Reply:
will revert to secondary since others is doing that too.